scripts/run_tests_one_by_one.py

```diff
@@ -75,7 +94,16 @@ def parse_assertions(stdout):
     if not profile:
         print(f"[{test_number}/{test_count}]: {test_case}", end="")
     start = time.time()
-    res = subprocess.run([unittest_program, test_case], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    try:
+        res = subprocess.run(
+            [unittest_program, test_case], stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=timeout
+        )
+    except subprocess.TimeoutExpired as e:
+        print(" (TIMED OUT)", flush=True)
+        if not no_exit:
```

Comment:
I think here it's missing a `        return_code = 1` line to pass on information that process failed (due to timeout).

Reply:
I am not completely sure this makes sense, since you might also want to use timeout as a way to run only short tests, but I think it's better to be strict here (so that we notice them on CI as failures). Up to you obv

Reply:
Yea I agree, definitely a timeout is an error and should be marked as such 👍